src/Interceptor/AbstractInterceptor.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenClassrooms\ServiceProxy\Interceptor;
+
+use OpenClassrooms\ServiceProxy\Annotation\Annotation;
+use OpenClassrooms\ServiceProxy\Contract\AnnotationHandler;
+use OpenClassrooms\ServiceProxy\Contract\Exception\DuplicatedDefaultHandler;
+use OpenClassrooms\ServiceProxy\Contract\Exception\DuplicatedHandler;
+use OpenClassrooms\ServiceProxy\Contract\Exception\HandlerNotFound;
+use OpenClassrooms\ServiceProxy\Contract\Exception\MissingDefaultHandler;
+
+abstract class AbstractInterceptor
```

Comment:
Here as methods `public function getPrefixPriority(): int` and `public function getSuffixPriority(): int` are implemented and all subclasses implements `(Prefix|Suffix)Interceptor`,  why don't we implements both `SuffixInterceptor, PrefixInterceptor` interfaces in this abstract class and add the following methods as abstract : 

- `public abstract function prefix(Instance $instance): Response;`
- `public abstract function supportsPrefix(Instance $instance): bool;`
- `public abstract function suffix(Instance $instance): Response;`
- `public abstract function supportsSuffix(Instance $instance): bool;`

Reply:
Ok I can see below that there is a subclass that doesn't implements both interfaces, so I'm wondering why methods `public function getPrefixPriority(): int` and  `public function getSuffixPriority(): int` are implemented in this class except for some factorisation. 

Finally we can have subclasses which implements one of `(Prefix|Suffix)Interceptor` or both or why not another Interceptor interface, so here those methods are part of two distinct interfaces, I think they should be implemented in the real implementation even if the code is duplicated.

In other words why a "common" abstract class implement some methods (which can be unused) from different interfaces without implement these interfaces ?

Reply:
I see the issue, I used this as a shortcut to have priority defaults into 0, but I think you have a point, and it's better to force each implementation think about it's priority anyway.